test/core/event_engine/posix/poller_fork_test.cc

```diff
@@ -0,0 +1,527 @@
+// Copyright 2025 gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <gmock/gmock.h>
+#include <grpc/event_engine/event_engine.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/support/status.h>
+#include <gtest/gtest.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <array>
+#include <cerrno>
+#include <csignal>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <iterator>
+#include <memory>
+#include <optional>
+#include <queue>
+#include <string>
+#include <thread>
+#include <utility>
+#include <vector>
+
+#include "absl/cleanup/cleanup.h"
+#include "absl/debugging/stacktrace.h"
+#include "absl/debugging/symbolize.h"
+#include "absl/functional/any_invocable.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/substitute.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/event_engine/channel_args_endpoint_config.h"
+#include "src/core/lib/event_engine/posix_engine/posix_engine.h"
+#include "src/core/lib/event_engine/tcp_socket_utils.h"
+#include "src/core/util/wait_for_single_owner.h"
+#include "test/core/test_util/port.h"
+#include "test/core/test_util/test_config.h"
+
+namespace grpc_event_engine::experimental {
+
+std::string GetBacktrace() {
+  std::array<void*, 10> ptrs;
+  int frames = absl::GetStackTrace(ptrs.data(), ptrs.size(), 1);
+  std::vector<std::string> bt(frames);
+  for (int i = 0; i < frames; ++i) {
+    std::array<char, 150> frame{0};
+    std::string frame_str = absl::Symbolize(ptrs[i], frame.data(), frame.size())
+                                ? std::string(frame.data(), frame.size())
+                                : "<Boop>";
+    bt[i] = absl::Substitute("  $0 $1", i, frame_str);
+  }
+  return absl::StrJoin(bt, "\n");
+}
+
+namespace {
+
+thread_local bool is_in_scheduler = false;
```

Comment:
Can this be a private data member of `TestScheduler`?  Or is there some reason that won't work for a thread_local?